### PR TITLE
add the separate run for preprocessing during training

### PR DESCRIPTION
**Pull request by Anna050689**
add the separate run for preprocessing during training

### CODE REVIEW

File: src/syngen/ml/strategies/strategies.py [hunk 1]
```diff
@@ -128,20 +128,25 @@ def run(self, **kwargs):
...
                 epochs=kwargs["epochs"],
                 drop_null=kwargs["drop_null"],
                 row_limit=kwargs["row_limit"],
-                table_name=kwargs["table_name"],
+                table_name=table,
                 metadata_path=kwargs["metadata_path"],
                 print_report=kwargs["print_report"],
                 batch_size=kwargs["batch_size"],
             )
-            MlflowTracker().log_params(self.config.to_dict())
```
Comment (rioman) [inline src/syngen/ml/strategies/strategies.py:141]: Don't we need to log parameters any more?

File: src/syngen/ml/vae/models/dataset.py [hunk 2]
```diff
@@ -1093,4 +1094,6 @@ def pipeline(self) -> pd.DataFrame:
 
         self.fit(self.df)
 
+        MlflowTracker().end_run()
```
Comment (rioman) [inline src/syngen/ml/vae/models/dataset.py:1097]: Is this "end_run" corresponds to "start_run" in wrappers.py?

File: src/syngen/ml/vae/wrappers/wrappers.py [hunk 3]
```diff
@@ -243,6 +243,14 @@ def fit_on_df(
 
         self.optimizer = self._create_optimizer()
         self.loss_metric = self._create_loss()
+
+        MlflowTracker().start_run(
```
Comment (rioman) [inline src/syngen/ml/vae/wrappers/wrappers.py:247]: Is this "start_run" corresponds to "end_run" in dataset.py?

File: src/syngen/ml/strategies/strategies.py (hunk 1 above)
Comment (Anna050689) [reply src/syngen/ml/strategies/strategies.py:141]: These parameters like "epochs", "print_report", "batch_size", "row_limit" and etc. are added to the section "parameters" to bring more information about parameters set in the particular run.

File: src/syngen/ml/vae/wrappers/wrappers.py (hunk 3 above)
Comment (Anna050689) [reply src/syngen/ml/vae/wrappers/wrappers.py:247]: No, this run corresponds to the "end_run" in "__infer_tables" of the class Worker

**Review by rioman — APPROVED** [review]

**Pull request merged**